src/testrun.rs

```diff
@@ -14,6 +14,30 @@ pub enum Outcome {
     Skip,
 }
 
+#[pymethods]
+impl Outcome {
+    #[new]
+    fn new(value: String) -> Self {
+        let val = value.as_str();
+        match val {
+            "pass" => Outcome::Pass,
+            "failure" => Outcome::Failure,
+            "error" => Outcome::Error,
+            "skip" => Outcome::Skip,
+            _ => Outcome::Failure,
+        }
+    }
+
+    fn __str__(&self) -> String {
```

Comment:
This is pretty much the same as a dunder method in python ya?

Reply:
yes, see more here: https://pyo3.rs/v0.10.1/class#string-conversions